### ISSUE TITLE
iex session appears in os x terminal not in qemu

### ISSUE THREAD

**Issue by cleesmith** [nerves-project/nerves_system_qemu_arm#4]
The iex session appears in the os x terminal not in the qemu window, but the nerves logo does appear in qemu ... is the following `mix firmware` command the reason why?  
If yes, how to fix ?

```
mix firmware
** (Nerves.Env.Exception) Major version mismatch between host and target Erlang/OTP versions
  Host version: 19
  Target version: 18

This will likely cause Erlang code compiled for the target to fail in
unexpected ways. Install an Erlang OTP release that matches the target
version before continuing.

    lib/mix/tasks/nerves.loadpaths.ex:20: Mix.Tasks.Nerves.Loadpaths.run/1
    (mix) lib/mix/task.ex:296: Mix.Task.run_task/3
    (mix) lib/mix/task.ex:328: Mix.Task.run_alias/3
    (mix) lib/mix/task.ex:261: Mix.Task.run/2
    (mix) lib/mix/task.ex:319: Mix.Task.get_task_or_run/3
    (mix) lib/mix/task.ex:278: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:58: Mix.CLI.run_task/2
```

installation steps:

```
$ brew update
$ brew install erlang
$ brew install elixir
$ brew install fwup squashfs coreutils
$ mix local.hex
$ mix local.rebar
$ mix archive.install https://github.com/nerves-project/archives/raw/master/nerves_bootstrap.ez

$ mix nerves.new hello_nerves --target qemu_arm
$ cd hello_nerves && mix deps.get && mix firmware
$ fwup -a -d _images/qemu_arm/hello_nerves.img -i _images/qemu_arm/hello_nerves.fw -t complete
$ qemu-system-arm -M vexpress-a9 -smp 1 -m 256                         \
    -kernel _build/qemu_arm/dev/nerves/system/images/zImage            \
    -dtb _build/qemu_arm/dev/nerves/system/images/vexpress-v2p-ca9.dtb \
    -drive file=_images/qemu_arm/hello_nerves.img,if=sd,format=raw     \
    -append "console=ttyAMA0,115200 root=/dev/mmcblk0p2" -serial stdio \
    -net none
```

brew info:

```
brew info erlang
erlang: stable 19.0.2 (bottled), HEAD
brew info elixir
elixir: stable 1.3.2 (bottled), HEAD
brew info qemu
qemu: stable 2.6.0 (bottled), HEAD
```

Thanks

**Comment by fhunleth** [comment]
We changed the Erlang prompt to appear in the terminal so that qemu could be run headless. See b1f893ac107b13db0ec4530610698a558d84aece.

As for the `mix firmware` error, that's odd. Which version of `nerves_system_qemu_arm` are you using? The latest, v0.6.1, is using Erlang 19 so you shouldn't be getting a host/target mismatch error.

**Comment by cleesmith** [comment]
So the `[![Screenshot](https://i.imgur.com/9JEMMGEh.jpg)](https://i.imgur.com/9JEMMGE.jpg)` in the README.md needs updating ?

I have this in my mix.exs file:

```
  def deps do
    [{:nerves, "~> 0.3.0"},
     {:nerves_system_qemu_arm, "~> 0.4.0"}
    ]
  end
```

in `deps/nerves_system_qemu_arm/version`: `0.4.1`
or how do you tell which version of `nerves_system_qemu_arm` ?

Since the iex session does work should I just ignore the error ?
Also, is there a way to actually run the iex session in the qemu window ?
Thanks, again.

**Comment by artob** [comment]
@cleesmith, the README instructions and screenshot will be updated tomorrow.

BTW, that `{:nerves_system_qemu_arm, "~> 0.4.0"}` dependency looks problematic. It should probably specify 0.6.0.

**Comment by cleesmith** [comment]
On the README it might help newbies, like me, to mention that the iex session is remote ... kind of similar to ssh but for an elixir session.  Maybe mention some commands like `File.ls "/"`, `File.cwd`, or `System.cmd("pwd", [])` in the **Hello, Nerves!** section.

Do I even need `{:nerves_system_qemu_arm, "~> 0.4.0"}` in my mix.exs file ? I was just following the **Installation** section at the bottom of the README.  But that **dep** was not added by doing `mix nerves.new hello_nerves --target qemu_arm`.

My past experience has been **remaster**ing images using Tiny Core, QEMU, and python with all of those problems and headaches ... so this is a cool project.
Thanks.

**Comment by fhunleth** [comment]
I think that this issue probably should have been closed a while ago. 

Also, so every knows, we started moving the `qemu_arm` system into a form that was much more like the other systems. It now does the A/B partition swapping and supports "OTA" firmware updates, but as a result, quite a bit of the docs need updating. A start on that is #16. I haven't deleted any of the old docs since there's tons of useful information there. The old docs are all in a subdirectory now to avoid confusion until there's time to pull in and update what's still relevant.
